src/core/config/remote/remote_config_validator.ts

```diff
@@ -0,0 +1,140 @@
+      ...RemoteConfigValidator.validateRelays(components, k8),
+      ...RemoteConfigValidator.validateHaProxies(components, k8),
+      ...RemoteConfigValidator.validateMirrorNodes(components, k8),
+      ...RemoteConfigValidator.validateEnvoyProxies(components, k8),
+      ...RemoteConfigValidator.validateConsensusNodes(components, k8),
+      ...RemoteConfigValidator.validateMirrorNodeExplorers(components, k8),
+    ]);
+  }
+
+  private static validateRelays(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
+    return Object.values(components.relays).map(async component => {
+      try {
+        const pods = await k8.getPodsByLabel([constants.SOLO_RELAY_LABEL]);
+
+        // to return the generic error message
+        if (!pods.length) throw new Error('Pod not found');
+      } catch (e) {
+        RemoteConfigValidator.throwValidationError('Relay', component, e);
+      }
+    });
+  }
+
+  private static validateHaProxies(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
+    return Object.values(components.haProxies).map(async component => {
+      try {
+        const pod = await k8.getPodByName(component.name);
+
+        // to return the generic error message
+        if (!pod) throw new Error('Pod not found');
+      } catch (e) {
+        RemoteConfigValidator.throwValidationError('HaProxy', component, e);
+      }
+    });
+  }
+
+  private static validateMirrorNodes(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
+    return Object.values(components.mirrorNodes).map(async component => {
+      try {
+        const pods = await k8.getPodsByLabel(constants.SOLO_HEDERA_MIRROR_IMPORTER);
+
+        // to return the generic error message
+        if (!pods.length) throw new Error('Pod not found');
+      } catch (e) {
+        RemoteConfigValidator.throwValidationError('Mirror node', component, e);
+      }
+    });
+  }
+
+  private static validateEnvoyProxies(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
+    return Object.values(components.envoyProxies).map(async component => {
+      try {
+        const pod = await k8.getPodByName(component.name);
+
+        // to return the generic error message
+        if (!pod) throw new Error('Pod not found');
+      } catch (e) {
+        RemoteConfigValidator.throwValidationError('Envoy proxy', component, e);
+      }
+    });
+  }
+
+  private static validateConsensusNodes(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
+    return Object.values(components.consensusNodes).map(async component => {
+      try {
+        const pod = await k8.getPodByName(component.name);
+
+        // to return the generic error message
+        if (!pod) throw new Error('Pod not found');
+      } catch (e) {
+        RemoteConfigValidator.throwValidationError('Consensus node', component, e);
+      }
+    });
+  }
+
+  private static validateMirrorNodeExplorers(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
```

Comment:
validateMirrorNodeExplorer
```suggestion
  private static validateMirrorNodeExplorer(components: ComponentsDataWrapper, k8: K8): Promise<void>[] {
```